### DISCUSSION TITLE
Running bash script after music stop

### DISCUSSION THREAD

**Discussion by Orneeman** [mikebrady/shairport-sync#1643] [Q&A]
Hi ,
I want to run a bash script after the music end that will turn off my AV receiver:
`#!/bin/bash
echo "tx 45:36" | cec-client -s -d 1
` 
and in the config file
` run_this_after_play_ends = "./home/airplay/stop.sh"; `

the script is with x+ , but it does not work ,
can you help me please?

**Comment by mikebrady** [comment]
Thanks for this, but it's really an issue rather than a discussion point.

Anyway, remember that all commands must be specified with fully qualified paths -- my guess is that this is the problem. There's a piece about events under [ADVANCED TOPICS](https://github.com/mikebrady/shairport-sync/tree/development/ADVANCED%20TOPICS).

**Comment by oliv-j** [comment]
Did you get this working? I'm looking into doing something similar that turns power on / off via a shelly relay.
Perhaps try running a simple script that writes to a log file / text file to see if it is shairport not running the script, or a problem with the script itself?

**Reply by Orneeman** [reply]
Hi, 
ok the print to log is working, how can I know what happens during the script run?

**Reply by oliv-j** [reply]
Try running in Terminal and see what errors are throwing.  You may need to build in some error handling into the script, or write the errors to a file:
[error handling](https://stackoverflow.com/questions/64786/error-handling-in-bash)
[Redirect errors](https://unix.stackexchange.com/questions/132511/how-to-capture-error-message-from-executed-command)

**Reply by Orneeman** [reply]
same :( , no log ....
run_this_before_entering_active_state = "sudo /home/airplay/stop.sh 2> /home/airplay/log.txt";

**Reply by Orneeman** [reply]
even "sudo reboot" doesn't work...
